### PR TITLE
Create ign-ionic collection view and set up nightlies

### PR DESCRIPTION
**Pull request by azeey**
Part of https://github.com/gazebo-tooling/release-tools/issues/1027 and https://github.com/gazebo-tooling/release-tools/issues/1026

### CODE REVIEW

**Comment by azeey** [comment]
`xml_config_content_changed.diff`:

```diff
Only in /tmp/pr_xml_configuration: gz-cmake4-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-common6-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-fuel-tools10-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-gui9-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-ionic-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-launch8-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-math8-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-msgs11-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-physics8-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-plugin3-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-rendering9-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-sensors9-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-sim9-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-transport14-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz-utils3-debbuilder.xml
Only in /tmp/pr_xml_configuration: gz_ionic-install-pkg-jammy-amd64.xml
Only in /tmp/pr_xml_configuration: ign-ionic.xml
Only in /tmp/pr_xml_configuration: ign_ionic-ci-win.xml
Only in /tmp/current_xml_configuration: ignition-harmonic-nightly-scheduler.xml
Only in /tmp/pr_xml_configuration: ignition-ionic-nightly-scheduler.xml
Only in /tmp/pr_xml_configuration: ignition_ionic-ci-main-homebrew-amd64.xml
Only in /tmp/pr_xml_configuration: ignition_ionic-install-pkg_ros_bootstrap-any-manual.xml
Only in /tmp/pr_xml_configuration: ignition_ionic-install_bottle-homebrew-amd64.xml
diff -ur -I '.*<id>dashboard_portlet_.*</id>.*' /tmp/current_xml_configuration/sdformat14-debbuilder.xml /tmp/pr_xml_configuration/sdformat14-debbuilder.xml
--- /tmp/current_xml_configuration/sdformat14-debbuilder.xml	2023-10-12 18:28:31.391635962 +0000
+++ /tmp/pr_xml_configuration/sdformat14-debbuilder.xml	2023-10-12 18:26:49.118625151 +0000
@@ -146,7 +146,7 @@
         <hudson.tasks.Shell>
             <command>#!/bin/bash -xe
 
-
+export NEED_C17_COMPILER=true
 export ENABLE_ROS=false
 /bin/bash -x ./scripts/jenkins-scripts/docker/multidistribution-debbuild.bash
 </command>
Only in /tmp/pr_xml_configuration: sdformat15-debbuilder.xml

```

File: jenkins-scripts/dsl/gz-collections.yaml [hunk 1]
```diff
@@ -337,6 +337,83 @@ collections:
...
             distros:
               - jammy
+  - name: 'ionic'
+    libs:
+      - name: gz-cmake
+        major_version: 4
+        repo:
+          current_branch: main
+      - name: gz-tools
+        major_version: 2
```
Comment (iche033) [inline jenkins-scripts/dsl/gz-collections.yaml:347]: we're bumping tools right?

Comment (azeey) [reply jenkins-scripts/dsl/gz-collections.yaml:347]: No, as @scpeters  pointed out on slack, bumping gz-tools will prevent us from installing Ionic side by side with any older release, so the plan is to update gz-tools to build with either gz-cmake3 or gz-cmake4. I've updated the description in https://github.com/gazebo-tooling/release-tools/issues/1027

Comment (iche033) [reply jenkins-scripts/dsl/gz-collections.yaml:347]: ok plan sounds good

**Review by iche033 — APPROVED** [review]

**Pull request merged**